addons/api/addon/serializers/credential.js

```diff
@@ -2,20 +2,43 @@ import ApplicationSerializer from './application';
 import { copy } from 'ember-copy';
 
 export default class CredentialSerializer extends ApplicationSerializer {
-  serialize() {
+  /**
+   * @override
+   * @method serialize
+   * @param {Snapshot} snapshot
+   */
+  serialize(snapshot) {
+    switch (snapshot.record.type) {
+      case 'username_password':
+        return this.serializeUsernamePassword(...arguments);
+      case 'ssh_private_key':
+      default:
+        return this.serializeSSHPrivateKey(...arguments);
+    }
+  }
+
+  serializeUsernamePassword() {
     const serialized = super.serialize(...arguments);
+    // Remove password_hmac and non-username_password type attributes
```

Comment:
I'm wondering if it makes sense to pick the fields we actually want for each case rather than deleting the ones we don't want? Doing the latter puts us in the situation of having to add more things to delete for each case when a new credential type is added

Reply:
I had a similar thought but not sure the best way to accomplish that. My biggest issue is it would be a lot of duplication for the fields like name, desc, type, and username. After some testing, we actually don't need to delete the hmac attributes because they don't exist on the model. I will remove those shortly. For the other attributes, I am not entirely sure the best method.

Reply:
That's true, it would be pretty annoying as well.

I was going to suggest that maybe we should use an attribute like `isSecret` to not serialize them like we do with `readOnly` fields but I notice that we have a similar [one that exists](https://github.com/hashicorp/boundary-ui/blob/ICU-5646-serializer/addons/api/addon/serializers/application.js#L60) already to prevent normalization, but I don't see it preventing serialization (not sure what happens the secrets object when serializing). 

Perhaps this is something we could use or do something similar on the serialization side?

Reply:
This sounds like more great content for our FE Techtime. Thanks for the feedback though. It is great to discuss this stuff.